src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -3501,6 +3539,18 @@ static QVariant fcnArrayGet( const QVariantList &values, const QgsExpressionCont
   return list.at( pos );
 }
 
+static QVariant fcnArrayFirst( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent )
+{
+  const QVariantList list = QgsExpressionUtils::getListValue( values.at( 0 ), parent );
+  return list.value( 0 );
```

Comment:
Is this one not enough? https://github.com/qgis/QGIS/pull/4825/files#diff-7df745f17644684d537a831b9257dccfR1133

`value()` already checks for non empty list. But I now remember that you don't like it for some reason ;)